.github/container/test-pax.sh

```diff
@@ -0,0 +1,264 @@
+    echo "  -o, --output NAME          Name for the output folder, a temporary folder will be created if none specified."
+    echo "  --data-parallel            Data parallelism to use. Defaults to 1."
+    echo "  --tensor-parallel          Tensor parallelism to use. Defaults to 1."
+    echo "  --pipeline-parallel        Pipeline parallelism to use. Defaults to 1 for no pipelining." 
+    echo "  -h, --help                 Print usage."
+    exit $1
+}
+
+args=$(getopt -o b:s:o:h --long batch-per-gpu:,dtype:,steps:,help,multiprocess,output:,data-parallel:,tensor-parallel:,pipeline-parallel: -- "$@")
+if [[ $? -ne 0 ]]; then
+    exit $1
+fi
+
+# Default arguments
+MULTIPROCESS=0
+OUTPUT=$(mktemp -d)
+
+export BATCH_PER_GPU=4
+export DTYPE='bfloat16'
+export STEPS=500
+export DP=1
+export TP=1
+export PP=1
+export VOCAB_PATH="gs://t5-data/vocabs/cc_all.32000.100extra/sentencepiece.model"
+
+eval set -- "$args"
+while [ : ]; do
+    case "$1" in
+        -b | --batch-per-gpu)
+            export BATCH_PER_GPU="$2"
+            shift 2
+            ;;
+        --dtype)
+            export DTYPE="$2"
+            shift 2
+            ;;
+        -s | --steps)
+            export STEPS="$2"
+            shift 2
+            ;;
+        --multiprocess)
+            MULTIPROCESS=1
+            shift 1
+            ;;
+        -o | --output)
+            OUTPUT=$2
+            shift 2
+            ;;
+        --data-parallel)
+            export DP="$2"
+            shift 2
+            ;;
+        --tensor-parallel)
+            export TP="$2"
+            shift 2
+            ;;
+        --pipeline-parallel)
+            export PP="$2"
+            shift 2
+            ;;
+        -h | --help)
+            usage 1
+            ;;
+        --)
+            shift;
+            break 
+            ;;
+        *)
+            echo "UNKNOWN OPTION $1"
+            usage 1
+    esac
+done
+
+# # Set derived variables
+
+export NGPUS=$(nvidia-smi -L | grep -c '^GPU')
+
+print_var BATCH_PER_GPU
+print_var DTYPE
+print_var STEPS
+print_var NGPUS
+print_var OUTPUT
+print_var MULTIPROCESS
+print_var DP
+print_var TP
+print_var PP
+
+## Enter Paxml source folder
+### TODO: check this.. not quite sure what's happening here
+## getting paxml path? I guess we are just using python import to find where paxml is located
+
+## TODO: figure out how to disable checkpoint saving altogether!! 
+## do not want to save checkpoint 0!
+PAXML_DIR=$(dirname `python -c 'import paxml; print(*paxml.__path__)'`)
+pushd ${PAXML_DIR}
+
+## Create configs file
+cat > ci_configs.py <<EOF
+import os
+import math
+from paxml import tasks_lib, experiment_registry
+from paxml.contrib.gpu.scripts_gpu.configs import GPT126M, configure_gpt3_task
+from paxml.tasks.lm.params.c4 import TransformerLmSpmdPipelineAdam
+from paxml.tasks.lm.params.lm_cloud import SyntheticDataset
+from praxis import base_layer
+from praxis import layers
+
+dp = int(os.getenv("DP"))
```

Comment:
Directly embedding the value using variable substitution may be better for archiving purposes. Otherwise, we won't know the value of `dp` without knowing the value of the environment variable `DP`, which is not readily available from this script.